c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -41,6 +41,11 @@ struct TypeChecker<'tcx, 'a> {
     equiv_constraints: Vec<(PointerId, PointerId)>,
 }
 
+fn is_castable_to(_t: LTy, _u: LTy) -> bool {
+    // TODO: implement
+    true
+}
```

Comment:
Is this meant to be something similar to what I implemented in #839 and #841?

Reply:
yes, the idea was that this is where the bulk of your checks in those PRs would take place